_automating-workflows/api/get-workflow-steps.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: default
+title: Get workflow steps
+parent: Workflow API
+nav_order: 50
+---
+
+# Get workflow steps
+
+OpenSearch validates workflows using the [JSON file](https://github.com/opensearch-project/automating-workflows/blob/main/src/main/resources/mappings/workflow-steps.json) that lists the required inputs and generated outputs for all steps. The Get Workflow Steps API retrieves this file.   
+
+## Path and HTTP methods
+
+```json
+GET /_plugins/_flow_framework/workflow/_steps
+``` 
+
+### Example request
+
+```json
+GET /_plugins/_flow_framework/workflow/_steps
+```
+{% include copy-curl.html %}
+
+
+### Example response
+
+OpenSearch responds with the [JSON file](https://github.com/opensearch-project/automating-workflows/blob/main/src/main/resources/mappings/workflow-steps.json) containing the steps. The order of fields in the returned steps may not exactly match the original JSON but will function identically.
```

Comment:
Similar comment here about a direct file link.  Do you think we should include the JSON from a sample workflow step showing what it contains (input, output, required plugins, and default step timeout)?  Such as
```json
{
    "register_remote_model": {
        "inputs": [
            "name",
            "connector_id"
        ],
        "outputs": [
            "model_id",
            "register_model_status"
        ],
        "required_plugins":[
            "opensearch-ml"
        ]
    }
}
```

Reply:
Good idea, I will delete the links to the file and add an example. The users will see the template once they call the API.